File: 6-Patron/php/DecoratorB_BotasJungla.php
<?php
include './FabricaDecorador_BendicionBotas.php';
include './Bendicion_BotasJungla.php';
class DecoratorB_BotasJungla extends FabricaDecorador_BendicionBotas{
	private $bendecirBotasJungla;

	public function __construct($bendecirBotasJungla,$botasNormales,$creaBotaAgregada){
    	$this->bendecirBotasJungla = $bendecirBotasJungla;
    	$this->botasNormales = $botasNormales;
    	$this->creaBotaAgregada = $creaBotaAgregada;
	}
 	
 	public function __construct(){
    	$this->bendecirBotasJungla = null;
    	$this->botasNormales = null;
    	$this->creaBotaAgregada = null;
	}
 	public function bendecirBotas($botas){
    	echo ' El dios de la jungla ha bendecido tus botas de ' . $botas->getVelocidad() . ' de velocidad';
 		$velocidadNueva = $botas->getVelocidad() + 20;
 		$botas->setVelocidad($velocidadNueva);
 		echo ' a ' . $velocidadNueva() . ' de velocidad y obtuviste un bono de daño jungla';
    	return $botas;
	}

	public function crearBotas(){
    	$nuevaBotaJ = new Bendicion_BotasJungla('Bendición del dios de la jungla');
    	$this->bendecirBotasRecorrer = $nuevaBotaJ;
	}
}
?>

File: 6-Patron/python/DecoratorB_BotasJungla.py
import FabricaDecoradorBendicionBotas
import BendicionBotasJungla


class DecoratorB_BotasJungla(FabricaDecoradorBendicionBotas):
    def __init__(self, bendecirBotasJungla):
        super().__init__()
        self.__bendecirBotasJungla = bendecirBotasJungla

    def bendecirBotasJungla(self, botas):
        return self.__bendecirBotasJungla

    def crearBotas(self):
        self.__bendecirBotasJungla = BendicionBotasJungla(
            'Bendición del dios de la jungla')
